### PR TITLE
add ImageCRDBuilder implement

### PR DESCRIPTION
**Pull request by cdjianghan**
add some imgeCrdBuilder change
    add CheckLoginStatus() to check wheather push image should appen
    ReadOrBuildAppConfig() add logic when user didn't write any image config
    add Image base info read such as image size image arch ...
    delete action content read now

### CODE REVIEW

**Comment by fanux** [comment]
文件名统一下划线加小写

File: pkg/buildah/imageCrdBuilder.go [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+}
+
+type dockerConfigFile struct {
+	AuthConfigs map[string]dockerAuthConfig `json:"auths"`
+	CredHelpers map[string]string           `json:"credHelpers,omitempty"`
+}
+
+func NewAndRunImageCRDBuilder(args []string) {
+	//run without error returns
+	icb := &ImageCRDBuilder{args[0], strings.Split(args[0], "/")[0], "", "", "", &imagev1.Image{}}
```
Comment (fanux) [inline pkg/buildah/imageCrdBuilder.go:91]: What if image name not contains "/"

Comment (cdjianghan) [reply pkg/buildah/imageCrdBuilder.go:91]: if user don't specified image repo,eee
what about use hub.sealos.cn as default?

**Review by fanux — APPROVED** [review]

File: pkg/buildah/imageCrdBuilder.go [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+	return ReadOrBuildAppConfigOutput, nil
+}
+
+func (icb *ImageCRDBuilder) ConfigParse(c []byte) error {
+	config := imagev1.Image{}
+	_, _, err := yaml.NewDecodingSerializer(unstructured.UnstructuredJSONScheme).Decode(c, nil, &config)
+	if err != nil {
+		return fmt.Errorf("configParse err : %v", err)
+	}
+	config.Spec.DetailInfo.Actions = make(map[string]imagev1.Action)
```
Comment (zzjin) [inline pkg/buildah/imageCrdBuilder.go:264]: Actions? Split out?

File: pkg/buildah/imageCrdBuilder.go (hunk 1 above)
Comment (zzjin) [reply pkg/buildah/imageCrdBuilder.go:91]: Do not split and parse image names by hand ,using @lingdie's interface function.

File: pkg/buildah/imageCrdBuilder.go [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+	err = json.Unmarshal(encodeAuth, &dockerConfig)
+	if err != nil {
+		return false
+	}
+	//auth struct is username:password which is namespcae:kubeconfig in sealos.hub scenario
+	auth, err := base64.StdEncoding.DecodeString(dockerConfig.AuthConfigs[icb.repositoryname].Auth)
+	if err != nil {
+		return false
+	}
+	icb.username = strings.SplitN(string(auth), ":", 2)[0]
```
Comment (zzjin) [inline pkg/buildah/imageCrdBuilder.go:121]: same as above?

Comment (cdjianghan) [reply pkg/buildah/imageCrdBuilder.go:121]: ok

Comment (cdjianghan) [reply pkg/buildah/imageCrdBuilder.go:121]: I double check this code ,It different from above .It read auth.json ,the struct in this file is different with simple docker domin parse,so this code I think better remain.

**Comment by cdjianghan** [comment]
**Local test output :**

**test case 1 :**
root@VM-12-8-ubuntu:~/sealos# sealos push f161ad13fdf0 hub.sealos.cn/aaa/bb:cc
Getting image source signatures
Copying blob 55f42b62daac skipped: already exists  
Copying config f161ad13fd done  
Writing manifest to image destination
Storing signatures
start ImageCrd Push
Success create container
Success get appConfig
Success read image content
Success read readme file
Success image apply

**Review by fanux — APPROVED** [review]

**Pull request merged**